packages/opentelemetry-types/src/distributedcontext/entry.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * An Entry is used to label anything that is associated with a specific
+ * operation, such as an HTTP request.
+ */
+export interface Entry {
```

Comment:
This API is too complicated for no reason and should be simplified. I'm starting to see a similar over-engineering pattern like what happened in OpenCensus.

Reply:
How will this be used? Is it a public or private API?

Reply:
> How will this be used? Is it a public or private API?

There will be a `DistributedContext` public API which will have below operations:
get (params: `EntryKey`)
put (params: `EntryKey, EntryValue, EntryMetadata` (optional)), 
remove (params: `EntryKey`).

> This API is too complicated for no reason and should be simplified. I'm starting to see a similar over-engineering pattern like what happened in OpenCensus.

What's your suggestion?

Reply:
To be compliant with the API described above, I would do something like this:

```ts
export interface Entry {
  readonly key: string;
  readonly value: string;
  readonly ttl: number;
}
```

No need for a more complicated structure.

However, as described above, I would use 3 different APIs. The use cases you mentioned above are completely different and should be handled in different ways, which will only result in more and more metadata being added to `Entry` since it's trying to do too much.

For example, a simple `setBaggageItem(key, value)` similar to OpenTracing could solve the trace context use case.

Regular injection and extraction can handle a single hop.

No construct is needed for local propagation since this is already handled by the `SpanContext`.

Reply:
Thanks for the feedback, I have simplified the key and value types. In the future, additional properties may be added to `EntryMetadata` type. Hence, would like to keep as it is. 

Specs for `EntryMetadata`: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/distributedcontext-api.md#entrymetadata.

Reply:
I opened https://github.com/open-telemetry/opentelemetry-specification/issues/140 with an alternative API proposal.

@mayurkale22 Is this something that could be implemented right away if you think it makes sense? Having a reference implementation for this may help showcase the advantages of this approach.